app/controllers/osbbs_controller.rb

```diff
@@ -0,0 +1,54 @@
+class OsbbsController < ApplicationController
+  before_action :obtain_osbb, only: %i[show edit update destroy]
+
+  def index
+    @osbbs = Osbb.all
+  end
+
+  def new
+    @osbb = Osbb.new
+  end
+
+  def show; end
+
+  def create
+    @osbb = Osbb.new(osbb_params)
+    if @osbb.save
+      flash[:success] = 'Osbb profile created'
+      redirect_to @osbb
+    else
+      render 'new'
+    end
+  end
+
+  def edit; end
+
+  def update
+    if @osbb.update(osbb_params)
+      flash[:success] = 'Osbb profile updated'
+      redirect_to @osbb
+    else
+      render 'edit'
+    end
+  end
+
+  def destroy
+    @osbb.destroy
+    redirect_to [:osbbs]
+    flash[:success] = 'Osbb profile deleted'
+  end
+
+  private
+
+  def osbb_params
+    params.require(:osbb).permit(:name,
+                                 :phone,
+                                 :email,
+                                 :website,
+                                 :is_available)
+  end
+
+  def obtain_osbb
+    @osbb = Osbb.find(params[:id])
```

Comment:
use memoize 
https://www.justinweiss.com/articles/4-simple-memoization-patterns-in-ruby-and-one-gem/